### PR TITLE
More lighting work.

### LINKED ISSUE #4869: Space can be renamed
I can call it "lelfaggetz" or some other shit. Occurred after the new lighting merge.

### LINKED ISSUE #4882: (WEB REPORT BY: notmog REMOTE: 198.245.63.50:7777) AI Upload Turrent Controls Broken
>Revision (Should be above if you're viewing this from ingame!)
4e6d381df0ee0f356ce39a870fb7f679575d8cd0

>General description of the issue
AI Upload turrets do not deactivate when they are disabled on the turret control.

>What you expected to happen
Turrets disabled.

>What actually happened
Turrets were actually still active.

>Steps to reproduce if possible
Try using the AI Upload turret control.

### LINKED ISSUE #4895: Areas lack light because they are not lit physically
Some places are dark as shit when they should be lit
This is more of a list
Its not a map issue and more of a make the area lit forcibly
So far

>Syndie shuttle, maybe other shuttles too, and where they land on the station or around it. 
>Also vox shuttle with same implications

### PR DESCRIPTION
**Pull request by PJB3005**
- Fixes #4869
- Makes `apply_lum()` and `remove_lum()` go by if a turf ACTUALLY has an overlay instead if it should, because how it was currently implemented didn't take the area's `lighting_use_dynamic` into account, and at that point there were 3 goddamn checks in those `if()` statements so I just used what we have, instead of what we _should_ have.
- Also gives shuttle turfs `dynamic_lighting = 0`, because they _really_ should.
- Fixes #4882
- Fixes #4895 
- More optimizations.
- Fixes some pooling issues.
- Basically reverts #4886, it broke a lot of shit.
- Adds extra logic & math to `update_lumcount()` to make it only cause updates if it would actually matter.

### CODE REVIEW

**Comment by 9600bauds** [comment]
I don't think you linked the right PR

**Comment by PJB3005** [comment]
Whoops, I didn't indeed
